### ISSUE TITLE
Truncation works incorrectly when handling duplicate detection

### ISSUE THREAD

**Issue by cdown** [cdown/clipmenu#67]
When we do duplicate detection, we mess up how much we should truncate by, resulting in this kind of mess

```
1520371520378510285060627 foo
1520378511823497558 22 (3 lines)
1520378511950295150 |current)',
1520378512000104671 urrent)',
1520378512049700306 rrent)',
1520378512083840324 rent)',
1520378512233257672 nt)',
152037867191520379302154021604 bar

```

**Comment by cdown** [comment]
I so far didn't manage to repro, but something funky is going on here

**Comment by cdown** [comment]
Ah shit, I think I worked it out.

`[[ ${last_data[any]} != "$data" ]]` is checked after we already set last_data for a particular selection. This means that we *might not* actually write even though we got to the final stage if we determine (say) the data we now have on CLIPBOARD was already ingested from PRIMARY, or vice-versa. By this time, we already set `$cache_file_output`, which we then move into `last_cache_file_output[$selection]`, and we then truncate on duplicate detection, but critically we may never have actually written the thing we want to truncate, because we know it's already in the other selection.
